src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -316,6 +317,12 @@ public DbConnection Close()
 		/// </summary>
 		public override void AfterTransactionCompletion(bool success, ITransaction tx)
 		{
+			if (!_transactionCoordinatorShared)
+				foreach (var sharingSession in ConnectionManager.SessionsSharingManager)
+				{
+					sharingSession.AfterTransactionCompletion(success, tx);
```

Comment:
I could have put that in callers, but then it was duplicated between `AdoTransaction` and `AdoNetWithDistributedTransactionFactory`.